lib/install/validate/validator.go

```diff
@@ -357,6 +359,86 @@ func (v *Validator) generateBridgeName(ctx, input *data.Data, conf *metadata.Vir
 	return input.DisplayName
 }
 
+func (v *Validator) firewall(ctx context.Context) {
+	defer trace.End(trace.Begin(""))
+
+	var hosts []*object.HostSystem
+	var err error
+
+	rule := types.HostFirewallRule{
+		Port:      8080, // serialOverLANPort
+		PortType:  types.HostFirewallRulePortTypeDst,
+		Protocol:  string(types.HostFirewallRuleProtocolTcp),
+		Direction: types.HostFirewallRuleDirectionOutbound,
+	}
+
+	if hosts, err = v.Session.Datastore.AttachedClusterHosts(ctx, v.Session.Cluster); err != nil {
+		log.Errorf("Unable to get the list of hosts attached to given storage: %s", err)
+		v.NoteIssue(err)
+		return
+	}
+
+	var misconfiguredEnabled []string
+	var misconfiguredDisabled []string
+	var correct []string
+
+	for _, host := range hosts {
+		fs, err := host.ConfigManager().FirewallSystem(ctx)
+		if err != nil {
+			v.NoteIssue(err)
+			return
+		}
+
+		disabled := false
+		esxfw, err := esxcli.GetFirewallInfo(host)
+		if err == nil && !esxfw.Enabled {
+			disabled = true
+			log.Infof("Firewall status: DISABLED on %s", host.InventoryPath)
+		} else {
+			log.Infof("Firewall status: ENABLED on %s", host.InventoryPath)
```

Comment:
Does this mean if we got error here, we'll assume the firewall is enabled?

Can we warn user for this error, even if we don't stop installation? Or maybe if forced, we continue, otherwise, just stop here?